beta/src/content/apis/react-dom/server/renderToPipeableStream.md

```diff
@@ -2,22 +2,576 @@
+          <Posts />
+        </Suspense>
+      </Suspense>
+    </ProfileLayout>
+  );
+}
+```
+
+In this example, React can start streaming the page even earlier. Only `ProfileLayout` and `ProfileCover` must finish rendering first because they are not wrapped in any `<Suspense>` boundary. However, if `Sidebar`, `Friends`, or `Photos` need to load some data, React will send the HTML for the `BigSpinner` fallback instead. Then, as enough data becomes available, more content will continue to be revealed until all of it becomes visible.
+
+Streaming does not need to wait for React itself to load in the browser, or for your app to become interactive. The HTML content from the server will get progressively revealed before any of the `<script>` tags have loaded.
+
+[Read more about how streaming HTML works.](https://github.com/reactwg/react-18/discussions/37)
+
+<Note>
+
+**Only Suspense-enabled data sources will activate the Suspense component.** They include:
+
+- Data fetching with Suspense-enabled frameworks like [Relay](https://relay.dev/docs/guided-tour/rendering/loading-states/) and [Next.js](https://nextjs.org/docs/advanced-features/react-18)
+- Lazy-loading component code with [`lazy`](/apis/react/lazy)
+
+Suspense **does not** detect when data is fetched inside an Effect or event handler.
+
+The exact way you would load data in the `Posts` component above depends on your framework. If you use a Suspense-enabled framework, you'll find the details in its data fetching documentation.
+
+Suspense-enabled data fetching without the use of an opinionated framework is not yet supported. The requirements for implementing a Suspense-enabled data source are unstable and undocumented. An official API for integrating data sources with Suspense will be released in a future version of React. 
+
+</Note>
+
+---
+
+### Specifying what goes into the shell {/*specifying-what-goes-into-the-shell*/}
+
+The part of your app outside of any `<Suspense>` boundaries is called *the shell:*
+
+```js {3-5,13,14}
+function ProfilePage() {
+  return (
+    <ProfileLayout>
+      <ProfileCover />
+      <Suspense fallback={<BigSpinner />}>
+        <Sidebar>
+          <Friends />
+          <Photos />
+        </Sidebar>
+        <Suspense fallback={<PostsGlimmer />}>
+          <Posts />
+        </Suspense>
+      </Suspense>
+    </ProfileLayout>
+  );
+}
+```
+
+It determines the earliest loading state that the user may see:
+
+```js {3-5,13
+<ProfileLayout>
+  <ProfileCover />
+  <BigSpinner />
+</ProfileLayout>
+```
+
+If you wrap the whole app into a `<Suspense>` boundary at the root, the shell will only contain that spinner. However, that's not a pleasant user experience because seeing a big spinner on the screen can feel slower and more annoying than waiting a bit more and seeing the real layout. This is why usually you'll want to place the `<Suspense>` boundaries so that the shell feels *minimal but complete*--like a skeleton of the entire page layout.
```

Comment:
@gaearon @sebmarkbage 

I don't think this is supported actually. You can't really put a Suspense boundary around the <html> tag because the placeholder html will force the HTML parser into body mode before the actual <html> tag is parsed. and when you swap the real content streams in the <html> and <head> tags it contains will get swallowed.

We probably need to indicate that the highest you can put a Suspense Boundary is as a child of `<head>` or `<body>`

Reply:
Hmm yeah by “root” here I meant inside body but I can see how that’s confusing